### DISCUSSION TITLE
Set declarationKind and useImplementingTypes on client preset

### DISCUSSION THREAD

**Discussion by avin-kavish** [dotansimha/graphql-code-generator#9342] [Q&A]
I'd like to switch to interfaces and use implementing types with the client preset. I tried passing options like this, it didn't work,

```ts
    './gql/': {
      preset: 'client',
      config: {
        declarationKind: 'interface',
      },
    },
```

How do I set this?

**Comment by JoepLijnen** [comment]
I don't think this is currently supported (at least not for Vue, which I am using: https://the-guild.dev/graphql/codegen/docs/guides/react-vue#config-api ) . I desperately would like this functionality too; I now have to write very awkward code, seeing as the types that are generated are simply not compatible with calls to Typescript stuff such as `instanceof`.

This is actually supported when not using a client-preset   (as per, for example: https://stackoverflow.com/a/64322143 )  .. and it's painful to be lacking this. I seriously don't see why anyone would want to use `type` instead of `interface` in Typescript, except for situations where you are explicitly generating code for compatibility with Javascript.
